clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -199,20 +200,21 @@ public PollResult(final long timeMsTillNextPoll, final List<UnsentRequest> unsen
     public static class UnsentRequest {
         private final AbstractRequest.Builder<?> requestBuilder;
         private final FutureCompletionHandler handler;
-        private Optional<Node> node; // empty if random node can be choosen
+        private final Optional<Node> node; // empty if random node can be chosen
         private Timer timer;
 
         public UnsentRequest(final AbstractRequest.Builder<?> requestBuilder, final Optional<Node> node) {
-            this(requestBuilder, node, new FutureCompletionHandler());
+            Objects.requireNonNull(requestBuilder);
+            this.requestBuilder = requestBuilder;
+            this.node = node;
+            this.handler = new FutureCompletionHandler();
         }
 
         public UnsentRequest(final AbstractRequest.Builder<?> requestBuilder,
                              final Optional<Node> node,
-                             final FutureCompletionHandler handler) {
-            Objects.requireNonNull(requestBuilder);
-            this.requestBuilder = requestBuilder;
-            this.node = node;
-            this.handler = handler;
+                             final BiConsumer<ClientResponse, Throwable> callback) {
```

Comment:
I'm stil not sure if this is the most organic way of passing a callback.  Suggestions?

Reply:
Which way are you preferring: the `FutureCompletionHandler` or the `BiConsumer`?

Reply:
the BiConsumer.